src/scripts/main.js

```diff
@@ -1,5 +1,28 @@
 'use strict';
+const wall = document.querySelector('.wall');
 
-document.addEventListener('click', (e) => {
-  // write code here
+wall.addEventListener('click', (e) => {
+  const spider = document.querySelector('.spider');
+  const spiderRect = spider.getBoundingClientRect();
+  const wallRect = wall.getBoundingClientRect();
+   const clickX = e.clientX - wallRect.left;
+   const clickY = e.clientY - wallRect.top;
+ 
+   let topPosition = clickY - (spiderRect.height / 2);
+   let leftPosition = clickX - (spiderRect.width / 2);
+
+  if (topPosition < 0) {
+    topPosition = 0;
+  } else if (topPosition > wallRect.height) {
+    topPosition = wallRect.height;
+  }
+
+  if (leftPosition < 0) {
+    leftPosition = 0;
+  } else if (leftPosition > wallRect.width) {
+    leftPosition = wallRect.width;
+  }
```

Comment:
The same issue exists with the spider's movement to the right. The spider can move outside of the wall's boundaries on the right side. This is because you are not subtracting the spider's width from the wall's width. This means that the spider's left position can be exactly at the wall's width, which would place the right side of the spider outside of the wall. To fix this, you should subtract the spider's width from the wall's width when setting the maximum left position.